src/server/qtquick/weventjunkman.cpp

```diff
@@ -0,0 +1,41 @@
+// Copyright (C) 2023 JiDe Zhang <zhangjide@deepin.org>.
+// SPDX-License-Identifier: Apache-2.0 OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+
+#include "weventjunkman.h"
+#include "wseat.h"
+#include "winputdevice.h"
+
+#include <QQuickWindow>
+
+WAYLIB_SERVER_BEGIN_NAMESPACE
+
+WEventJunkman::WEventJunkman(QQuickItem *parent)
+    : QQuickItem(parent)
+{
+    setAcceptedMouseButtons(Qt::AllButtons);
+}
+
+bool WEventJunkman::event(QEvent *event)
+{
+    switch(event->type()) {
+        using enum QEvent::Type;
+    case MouseButtonPress: Q_FALLTHROUGH();
+    case MouseButtonRelease: Q_FALLTHROUGH();
+    case KeyPress: Q_FALLTHROUGH();
+    case KeyRelease: {
+        auto e = static_cast<QInputEvent*>(event);
+        auto inputDevice = WInputDevice::from(e->device());
+        if (Q_UNLIKELY(!inputDevice))
+            return false;
+
+        auto seat = inputDevice->seat();
+        if (!seat)
+            return false;
+        return seat->filterUnacceptedInputEvent(window(), e);
```

Comment:
why need this while we have `QWlrootsRenderWindow::unacceptedEventFilter`?

Reply:
Because some events not ignore by QQuickWindow, so will missing some unaccepted events at WOutputRenderWindow::event. using the WEventJunkman, we can ensure the unaccepted mouse/touch events send to WSeat.